Tests/Marketplace/marketplace_services.py

```diff
@@ -565,6 +568,77 @@ def upload_to_storage(self, zip_pack_path, latest_version, storage_bucket, overr
+        return int(major), int(minor), int(rev)
+
+    def prepare_release_notes(self, index_folder_path):
+        """Need to implement the changelog.md parsing and changelog.json creation after design is
+        finalized.
+
+        """
+        task_status = False
+        try:
+            release_notes_dir = os.path.join(self._pack_path, Pack.RELEASE_NOTES)
+            if os.path.exists(os.path.join(index_folder_path, self._pack_name, 'changelog.json')):
+                print_warning(f"Found Changelog for: {self._pack_name}")
+                changelog_path = os.path.join(self._pack_path, Pack.CHANGELOG_JSON)
+                if os.path.exists(release_notes_dir):
+                    found_versions = []
+                    for filename in os.listdir(release_notes_dir):
+                        _version = filename.replace('.md', '')
+                        version = _version.replace('_', '.')
+                        found_versions.append(version)
+                    latest_release_notes = max(found_versions, key=self.major_minor_rev)
+                    print_color(f"Latest ReleaseNotes version is: {latest_release_notes}", LOG_COLORS.GREEN)
+                    if self._current_version != latest_release_notes:
+                        print_error(f"Version mismatch detected between current version: {self._current_version} "
+                                    f"and latest release notes version: {latest_release_notes}")
+                    else:
+                        with open(changelog_path, "r") as changelog_file:
+                            changelog = json.load(changelog_file)
+                            if latest_release_notes in changelog:
+                                print_error(f"Found existing release notes for version: {latest_release_notes} "
+                                            f"in the {self._pack_name} pack.")
+                        latest_rn_file = latest_release_notes.replace('.', '_')
+
+                        latest_rn_path = os.path.join(release_notes_dir, latest_rn_file + '.md')
+                        with open(latest_rn_path, 'r') as changelog_md:
+                            changelog_lines = changelog_md.read()
+                        version_changelog = {'releaseNotes': changelog_lines,
+                                             'displayName': latest_release_notes,
+                                             'released': datetime.utcnow().strftime(Metadata.DATE_FORMAT)}
+                        changelog[latest_release_notes] = version_changelog
+                        with open(os.path.join(self._pack_path, 'changelog.json'), "w") as f:
+                            json.dump(changelog, f)
+
+                else:
+                    print_warning(f"No ReleaseNotes directory was found for pack: {self._pack_name}")
+
+            else:
+                print_warning(f"No release notes found for: {self._pack_name}")
```

Comment:
This is edge case that should be handled properly. Most of packs are have initial version `1.0.0` and don't have `changelog.json`.  I think we need to discuss it.